### ISSUE TITLE
Consider showing the Travis status in the homu queue

### ISSUE THREAD

**Issue by Ms2ger** [servo/homu#12]

**Comment by carols10cents** [comment]
As someone who hasn't worked on homu before, (and I may be asking for a general code overview that exists elsewhere, if so please point me to it) what's involved in fixing this? That is, 

- Where should this be shown? A new column on pages like https://buildbot2.rust-lang.org/homu/queue/rust ? Where is the code for that page?
- Where should the request to the github status API happen?
- Do the results get stored in a database somewhere or are they requested on demand?

Basically, could someone (@frewsxcv perhaps?) write up what the requirements of this are, exactly, and what the high-level steps of working on this are?

**Comment by Eh2406** [comment]
The queue template is in https://github.com/servo/homu/blob/master/homu/html/queue.html
and is served from https://github.com/servo/homu/blob/master/homu/server.py#L808

Don't know the answers to the rest of your questions. For one thing I'm not sure which status we want to report. The Travis status of the original pr?

**Comment by Manishearth** [comment]
Storing the results in a db is a bunch of work. Doable, but work.

Having them get fetched via JS should not be hard. May hit rate limits.

**Comment by frewsxcv** [comment]
Yeah, I think the easiest route here is doing clientside requests to the [GitHub status API](https://developer.github.com/v3/repos/statuses/#list-statuses-for-a-specific-ref), or maybe even the Travis API. I'm not really familiar with either of their APIs, so not sure how feasible it really is.

**Comment by carols10cents** [comment]
@frewsxcv what exactly would you like shown where?

**Comment by frewsxcv** [comment]
I'm envisioning another column added to the table that indicates if CI is passing or not. Ideally the colors would be start so one could easily see at a glance which PRs are problematic.

**Comment by carols10cents** [comment]
> the colors would be start

?

**Comment by frewsxcv** [comment]
`s/start/stark/`

I NEED MORE ☕️☕️☕️☕️☕️☕️☕️

**Comment by Manishearth** [comment]
I personally was envisioning a small colored dot that you can hover over to get individual sub-CI statuses. Same thing you see on github, basically.

**Comment by frewsxcv** [comment]
@kennytm made this very cool alternative homu dashboard which shows travis status, among other things ✨ 

https://github.com/kennytm/borsholder
